src/discord/events/message/registerTeam.ts

```diff
@@ -0,0 +1,214 @@
+import Discord from 'discord.js';
+
+import { Config } from '../../../entities/config';
+import logger from '../../../logger';
+import { colors } from '../../constants';
+import { Team } from '../../../entities/team';
+
+class TeamInfo {
+  idArr: string[] = [];
+
+  name = '';
+
+  description = '';
+
+  table = 0;
+
+  user = '';
+
+  constructor(i: string) {
+    this.user = i;
+  }
+
+  addMembers(mem: string[]): void {
+    this.idArr = mem;
+  }
+
+  addUser(): void {
+    this.idArr.push(this.user);
+  }
+
+  addName(n: string): void {
+    this.name = n;
+  }
+
+  addDescription(d: string): void {
+    this.description = d;
+  }
+
+  addTable(t: number): void {
+    this.table = t;
+  }
+}
+
+let registrationInProgress = false;
+let decrement = false;
+let findName: Team;
+let findTable: Team;
+
+const promptArr = [
+  'Who are you hacking with? You have 5 minutes to send a reply. If you are hacking alone, simply write `none`!\n**Be sure to input their Discord ID in a comma-separated list e.g. 111222333444555,666777888999000,...**',
+  "What's your team/app name? You have 2 minutes to answer.",
+  "What's your table number (e.g. 42)? You have 2 minutes to answer.",
+  'What does your project do? How will it make a difference? What technologies are used? You have 5 minutes to answer.',
+];
+const componentArr = ['members', 'name', 'table', 'description'];
+const timeArr = [300000, 120000, 120000, 300000];
+const currentUsers: string[] = [];
+
+async function question(
+  m: Discord.Message,
+  f: (m: Discord.Message) => boolean,
+  prompt: string,
+  t: number,
+  component: string,
+  currentTeam: TeamInfo,
+): Promise<void> {
+  await m.author.send(prompt);
+  await m.channel
+    .awaitMessages(f, {
+      max: 1,
+      time: t,
+    })
+    .then(async (collected) => {
+      const input = collected.first().content;
+      if (input === '!quit') {
+        registrationInProgress = false;
+        m.author.send('You successfully halted the signup process! Feel free to keep using the bot like normal.');
+      } else {
+        switch (component) {
```

Comment:
Assuming @johnkahn is on board with us refactoring to avoid the listener pattern, we should store a variable (`step` maybe?) in our `discordContext` for the user and simply use that to figure out how to react.